cohortextractor/measure.py

```diff
@@ -56,3 +55,10 @@ def calculate(self, data):
         result["value"] = result[self.numerator] / result[self.denominator]
 
         return result
+
+
+def _drop_duplicates(lst):
+    """
+    Preserves the order of the list.
+    """
+    return pandas.Series(lst).drop_duplicates().tolist()
```

Comment:
Ah, this is very interesting! So my instinct here is to cry out in horror, "No, you're invoking the entire Pandas machinery with all it's weird quirks just to remove duplicates from a list!". But clearly yours is different. Maybe something like "Why are you using this cryptic Python one-liner when there's a well named Pandas function which does this?" :smile: 

In a practical sense it's neither here nor there as this module is fundamentally "pandas code" whether we import it or not, but I find the programming psychology interesting and possibly evidence of having been bitten by different sorts of issue in the past.

Pulling cryptic one-liners out into their own function is never a bad idea though, even though I sometimes can't quite be bothered when it's only used once.

Reply:
I was in two minds about this. It was something of an experiment and I wouldn't have done it this way in my own codebase. I'm feeling my way here and I thought that the "all in on Pandas" vibe in the codebase was stronger than the "lots of small specialized functions" vibe.

I'm very happy to go the other way. I'll definitely pull it out into a function, though.

(I was also influenced by the fact that the StackOverflow question on this includes the Pandas approach as an answer. :-))

Reply:
> I thought that the "all in on Pandas" vibe in the codebase was stronger than the "lots of small specialized functions" vibe.

This is good point. I'm used to working with researchers who've learned Pandas alongside learning Python and have an (understandable) tendency to always reach for Pandas solutions even when they're wildly inappropriate, so I have a slightly allergic reaction to it sometimes. In this case though, given that we have to explicitly cast the argument to a Pandas type and then convert it back it again it still feels like a non-natural use of Pandas to me.